File: Android/DictionaryRobot/app/src/main/java/com/example/ramsr/dictionaryrobot/model/RobotResponse.java
package com.example.ramsr.dictionaryrobot.model;

public class RobotResponse {

    private String status;
    private String currentTerm;
    private String currentTermDefinition;
    private Boolean hasNextPage;
    private Boolean hasPreviousPage;
    private Boolean hasNextTerm;
    private Boolean hasPreviousTerm;

    public RobotResponse(String status, String currentTerm, String currentTermDefinition, Boolean hasNextPage, Boolean hasNextTerm, Boolean hasPreviousTerm, Boolean hasPreviousPage) {
        this.status = status;
        this.currentTerm = currentTerm;
        this.currentTermDefinition = currentTermDefinition;
        this.hasNextPage = hasNextPage;
        this.hasNextTerm = hasNextTerm;
        this.hasPreviousTerm = hasPreviousTerm;
        this.hasPreviousPage = hasPreviousPage;
    }

    public String getStatus() {
        return status;
    }

    public String getCurrentTerm() {
        return currentTerm;
    }

    public String getCurrentTermDefinition() {
        return currentTermDefinition;
    }

    public Boolean getHasNextPage() {
        return hasNextPage;
    }

    public Boolean getHasNextTerm() {
        return hasNextTerm;
    }

    public Boolean getHasPreviousTerm() {
        return hasPreviousTerm;
    }

    public Boolean getHasPreviousPage() {
        return hasPreviousPage;
    }

}

File: iOS/Dictionary Robot/Dictionary Robot/Model/RobotResponse.swift
//
//  RobotResponse.swift
//  Dictionary Robot
//
//  Created by Ram Sri Charan on 10/13/18.
//  Copyright © 2018 Ram Sri Charan. All rights reserved.
//

import Foundation

//------------------------------------------------------------------------------
////////////////// Model to decode and store JSON Response  ////////////////////
//------------------------------------------------------------------------------

struct RobotResponse : Decodable {
    var status : String
    var currentTerm : String
    var currentTermDefinition : String
    var hasNextPage : Bool
    var hasPreviousPage : Bool
    var hasNextTerm : Bool
    var hasPreviousTerm : Bool
    
    init(status : String, currentTerm : String, currentTermDefinition : String, hasNextPage : Bool,
         hasPreviousPage : Bool, hasNextTerm : Bool, hasPreviousTerm : Bool)
    {
        self.status = status
        self.currentTerm = currentTerm
        self.currentTermDefinition = currentTermDefinition
        self.hasNextPage = hasNextPage
        self.hasPreviousPage = hasPreviousPage
        self.hasNextTerm = hasNextTerm
        self.hasPreviousTerm = hasPreviousTerm
    }
}
